### ISSUE TITLE
[XSS/XSX] RetroAchievements with badges enabled causes crash when viewing achievement list

### ISSUE THREAD

**Issue by TreeBranches** [libretro/RetroArch#11990]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

RetroAchievements with badges enabled causes crash when trying to view your achievement list

### Expected behavior

For it to not crash.

### Actual behavior

If badges are disabled, they can be viewed okay. If a game has less than a certain amount (<70ish) achievements and badges are enabled, they can be viewed once, but viewing them again will cause a crash.

### Steps to reproduce the bug

1. Play any game with achievements enabled, and achievement badges enabled.
2. View the achievement list.
3. Experience it crashing, otherwise go back a level.
4. View the achievement list again, and witness the crash.

### Bisect Results

This has always been a problem on Xbox One/Series consoles.

### Version/Commit
You can find this information under Information/System Information

You can find this information under Information/System Information

- RetroArch: 1.9.0
- Build Date: Jan 7 2021
- Compiler MSVC (1928) 64 bit

### Environment information

Xbox One/S/X and Xbox Series S/X consoles.

**Comment by LibretroAdmin** [comment]
@TheRhysWyrill does this still happen?

**Comment by TheRhysWyrill** [comment]
Nope, as someone that regularly uses RetroAchievements I can confirm that viewing the Achievement list causing a crash hasn't been a thing for as long as I've been using RA on Xbox.
